### ISSUE TITLE
See if `check_embedded_masker` can be used more systematically across estimators

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#4605]
check_embedded_masker runs some checks on the masker of an estimator but only for those found in the decoding / decomposition subpackages.

https://github.com/nilearn/nilearn/blob/f5114ee0b831530907e554a669427e70837029d2/nilearn/_utils/masker_validation.py#L13-L42

Yet very similar checks are run for GLM and other estimators.

See for example:

https://github.com/nilearn/nilearn/blob/f5114ee0b831530907e554a669427e70837029d2/nilearn/glm/first_level/first_level.py#L982-L989

It may be worth checking whether some refactoring cannot be done to reuse some of those checks across estimators.

Some of those 'repeats' may become more and more obivous as we start extending support for SurfaceImage data.

I may be missing something obvious why it cannot be done, so please let me know before I go down this rabbithole.

**Comment by bthirion** [comment]
I agree, but I don't think we need this in the very short term.

**Comment by Remi-Gau** [comment]
yeah I don't think this is urgent but may probably lead to better architecture overall.
